profiler/src/main/java/com/splunk/opentelemetry/profiler/context/SpanLinkage.java

```diff
@@ -41,11 +43,11 @@ String getTraceId() {
   }
 
   RecordedThread getRecordedThread() {
-    return recordedThread;
+    return recordedEvent == null ? null : recordedEvent.getThread();
   }
 
   Long getThreadId() {
```

Comment:
This method can return a primitive long now:
```suggestion
  long getThreadId() {
```

Reply:
Nah, the thread can still be null.